src/App.js

```diff
@@ -20,9 +24,34 @@ const [message, setMessage] = useState('');
 
   }, []);
 
+  const ProtectedRoute = ({ children }) => {
+    // Placeholder to actually check if user is authenticated
+    const useAuth = () => {
+      if (Math.random() > 0.5) {
+        return true;
+      }
+      return false;
+    }
+
+    const user = useAuth();
+    if (!user) {
+      // User is not authenticated
+      return <Navigate to="/login" />;
+    }
+    return children;
+  };
```

Comment:
Can you move this to its own file? Also can you add in a prop for required auth level (maybe requiredAuthLevel)? This would be set to either "user" or "anonymous". When set to user it should have the behavior it currently has. When set to anonymous (a non-logged in user) it should redirect to the home page instead.

Reply:
Thank you! I did all changes. Could you please also take a look when you will have  a time in my second ticket [[Create a general modal component#11](https://github.com/Code-the-Dream-School/bb-practicum-team1-front/issues/11)](https://github.com/orgs/Code-the-Dream-School/projects/10?pane=issue&itemId=23263085) and help me to figure out some points?